Fix lp:1659992 "Function over userstat tables cause mysql to crash" (5.5)

percona-ysorokin opened the pull request:
Some 'SELECT' statements over 'table_statistics' IS table used to call
'check_grant()' from 'fill_schema_table_stats()' with 'table_list' parameter
which did not include any element that would have 'next_global' pointing to
'thd->lex->query_tables_own_last'. This was a violation of 'check_grant()'
pre-conditions.

Fixed by backporting part of the 5.7
"wl#6960: Refactoring auth code: milestone 1, new physical file structure"
(commit 0ed74a1) which added additional 'NULL' check to 'TABLE_LIST'
iteration in 'check_grant()'.

Added 'main.percona_bug1659992' MTR test case.

percona-ysorokin commented:
Jenkins build link
http://jenkins.percona.com/view/PS%205.5/job/percona-server-5.5-param/1532/

laurynas-biveinis commented on mysql-test/t/percona_bug1659992.test:
```diff
@@ -0,0 +1,29 @@
+#
```
No need for a new separate testcase - please merge with main.percona_userstat, replacing header comment with "--echo #" one

laurynas-biveinis commented on mysql-test/t/percona_bug1659992.test:
```diff
@@ -0,0 +1,29 @@
+  IF @@global.version LIKE '5.7.%' THEN
+    SELECT v.variable_value FROM performance_schema.global_variables v WHERE v.variable_name = in_name INTO var_value;
+  ELSE
+    SELECT v.variable_value FROM information_schema.global_variables v WHERE v.variable_name = in_name INTO var_value;
```
IIRC INFORMATION_SCHEMA object names are in uppercase (PFS lowercase)

laurynas-biveinis commented on mysql-test/t/percona_bug1659992.test:
```diff
@@ -0,0 +1,29 @@
+
+--disable_result_log
+SELECT *
+  FROM information_schema.table_statistics s
```
Likewise

percona-ysorokin commented:
Updated Jenkins build link (as no code changes since the previous commit, just single test run)
http://jenkins.percona.com/view/PS%205.5/job/percona-server-5.5-param/1533/
